gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -188,10 +196,34 @@ namespace gtsam {
 
   /* ************************************************************************ */
   std::vector<double> DecisionTreeFactor::probabilities() const {
+    // Set of all keys
+    std::set<Key> allKeys(keys().begin(), keys().end());
+
     std::vector<double> probs;
-    for (auto&& [key, value] : enumerate()) {
-      probs.push_back(value);
-    }
+
+    // Go through the tree
+    this->apply([&](const Assignment<Key> a, double p) {
```

Comment:
const & ?

Reply:
I'd prefer if you don't use the lambda in place but create `op` using an auto and properly document what it does.

Reply:
Done